### PR TITLE
feat: make keyword filtering in completion provider configurable

### PR DESCRIPTION
**Pull request by cdietrich**

### CODE REVIEW

File: packages/langium/src/lsp/completion/completion-provider.ts [hunk 1]
```diff
@@ -442,6 +441,11 @@ export class DefaultCompletionProvider implements CompletionProvider {
         });
     }
 
+    protected isKeywordWorthyToPropose(context: CompletionContext, keyword: ast.Keyword): boolean {
```
Comment (msujew) [inline packages/langium/src/lsp/completion/completion-provider.ts:444]: Minor: Sounds a bit verbose and it probably makes sense to keep this more in line with the existing `filterCrossReference`.
```suggestion
    protected filterKeyword(context: CompletionContext, keyword: ast.Keyword): boolean {
```

**Review by msujew — COMMENTED** [review]
While you're at it: I've noticed that the `filterCrossReference` method doesn't accept a `CompletionContext` as a parameter. In it's current state, the method is kind of useless. Do you mind adding that as well?

File: packages/langium/src/lsp/completion/completion-provider.ts (hunk 1 above)
Comment (cdietrich) [reply packages/langium/src/lsp/completion/completion-provider.ts]: done

**Comment by cdietrich** [comment]
adapted both methods

**Review by msujew — APPROVED** [review]
Alright, thanks 👍

File: packages/langium/src/lsp/completion/completion-provider.ts [hunk 2]
```diff
@@ -442,6 +441,11 @@ export class DefaultCompletionProvider implements CompletionProvider {
         });
     }
 
+    protected filterKeyword(_context: CompletionContext, keyword: ast.Keyword): boolean {
```
Comment (spoenemann) [inline packages/langium/src/lsp/completion/completion-provider.ts:444]: The method name is ambiguous. In the JS context, "filter" is usually understood as include, not "filter out".

**Comment by cdietrich** [comment]
@spoenemann got a mail about the naming, but cannot find the comment

**Pull request merged**